xontrib_term_integrations/utils.py

```diff
@@ -1,47 +1,211 @@
 import sys
+import base64
 
 from xonsh.environ import Env
+from xonsh.built_ins import XSH
+
+envx = XSH.env or {}
 
 
 class Codes:
     # https://iterm2.com/documentation-escape-codes.html equivalents
-    ESC = "\x1b"
-    ST = "\x07"
-    OSC = ESC + "]"  # \x5d
-    CSI = ESC + "["
+    ESC = "\x1b"  # Escape, starts all the escape sequences
+    BEL = "\x07"  # Bell
```

Comment:
any reason to replace the ST with BEL ?

Reply:
BEL is used explicitly in the freedesktop r[spec docs](https://gitlab.freedesktop.org/Per_Bothner/specifications/blob/master/proposals/semantic-prompts.md), so that was the motivating factor

> OSC "133;L\007"
> OSC "133;A" options "\007"


iTerm docs states it's either/or, so it shouldn't matter there
> ST means either BEL (hex code 0x07) or ESC \\.

so the common ground between two specs is `BEL`

Reply:
Thanks understood!